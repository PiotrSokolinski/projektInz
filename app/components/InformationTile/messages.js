/*
 * InformationTile Messages
 *
 * This contains all the text for the InformationTile component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.InformationTile'

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the InformationTile component!',
  },
  upToDate: {
    id: `${scope}.upToDate`,
    defaultMessage: 'You are up to date!',
  },
})
