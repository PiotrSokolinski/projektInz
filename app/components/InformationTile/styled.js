import styled from 'styled-components'

import { Eye } from 'styled-icons/octicons/Eye'

import { Colors } from 'themes'

export const EyeIcon = styled(Eye)`
  color: ${Colors.ecstasy};
  cursor: pointer;
`

export const Container = styled.div`
  border: 1px solid ${Colors.edward};
  border-radius: 5px;
  margin: 25px;
  display: flex;
  align-items: center;
  padding: 5px 15px;
  justify-content: space-between;
`

export const InfoContainer = styled.div`
  display: flex;
  align-items: center;
`

export const Title = styled.div`
  font-size: 15px;
  margin-left: 25px;
`
