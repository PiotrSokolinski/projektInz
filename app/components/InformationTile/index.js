/**
 *
 * InformationTile
 *
 */

import React from 'react'
import PropTypes from 'prop-types'

import { FormattedMessage } from 'react-intl'
import messages from './messages'
import * as Styled from './styled'

const InformationTile = ({ icon: Icon, onOpen }) => (
  <Styled.Container>
    <Styled.InfoContainer>
      <Icon size="40" />
      <Styled.Title>
        <FormattedMessage {...messages.upToDate} />
      </Styled.Title>
    </Styled.InfoContainer>
    <Styled.EyeIcon size="22" onClick={onOpen} />
  </Styled.Container>
)

InformationTile.propTypes = {
  icon: PropTypes.object,
  onOpen: PropTypes.func,
}

export default InformationTile
