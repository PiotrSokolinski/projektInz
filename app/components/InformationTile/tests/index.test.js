/**
 *
 * Tests for InformationTile
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import { Eye } from 'styled-icons/octicons/Eye'

import InformationTile from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<InformationTile />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <InformationTile icon={Eye} />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <InformationTile icon={Eye} />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
