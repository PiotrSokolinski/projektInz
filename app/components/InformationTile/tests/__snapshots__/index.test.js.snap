// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationTile /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-ca47cf-1 ZkUxJ"
>
  <div
    class="styled__InfoContainer-ca47cf-2 ioAGxW"
  >
    <svg
      aria-hidden="true"
      class="StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      height="40"
      viewBox="0 0 16 16"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="styled__Title-ca47cf-3 lmCbnz"
    >
      <span>
        You are up to date!
      </span>
    </div>
  </div>
  <svg
    aria-hidden="true"
    class="styled__EyeIcon-ca47cf-0 jqmOuv StyledIconBase-sc-bdy9j4 jbPeIR"
    fill="currentColor"
    focusable="false"
    height="22"
    viewBox="0 0 16 16"
    width="22"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;
